controllers/greptimedbcluster/controller.go

```diff
@@ -138,21 +138,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
+	if err = cluster.SetDefaults(); err != nil {
+		r.Recorder.Event(cluster, corev1.EventTypeWarning, "SetDefaultValuesFailed", fmt.Sprintf("Set default values failed: %v", err))
+		return ctrl.Result{}, err
+	}
+
 	// Means the cluster is just created.
 	if len(cluster.Status.ClusterPhase) == 0 {
 		klog.Infof("Start to create the cluster '%s/%s'", cluster.Namespace, cluster.Name)
-
```

Comment:
Can you describe the bug and how to reproduce it in the PR? You can also create an issue. Reverting the original logic without any description is not a good practice.